### PR TITLE
lua/encoders: slack

### PR DESCRIPTION
**Pull request by nathanleiby**
https://clever.atlassian.net/browse/INFRA-687

Slack Encoder. Writes payload as message. Allows parameterizing channel and username.

### CODE REVIEW

File: lua/encoders/slack.lua [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+-- [HttpOutput]
+-- address = "https://hooks.slack.com/services/YOUR/SLACK/WEBHOOK"
+-- message_matcher = "<MESSAGE_MATCHER>"
+-- encoder = "SlackEncoder"
+
+function process_message()
+    local payload = read_message("Payload")
+
+    local slack_alert = {}
+    slack_alert["icon_emoji"] = ":ghost:"
```
Comment (azylman) [inline lua/encoders/slack.lua]: Maybe allow custom emoji too?

File: lua/encoders/slack.lua [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+-- address = "https://hooks.slack.com/services/YOUR/SLACK/WEBHOOK"
+-- message_matcher = "<MESSAGE_MATCHER>"
+-- encoder = "SlackEncoder"
+
+function process_message()
+    local payload = read_message("Payload")
+
+    local slack_alert = {}
+    slack_alert["icon_emoji"] = read_message("Fields[icon_emoji]")
+    slack_alert["username"] = read_message("Fields[username]")
```
Comment (mohit) [inline lua/encoders/slack.lua]: we should not have them as fields but rather as config?

Comment (nathanleiby) [inline lua/encoders/slack.lua]: Changed to config 82aeb9e

File: lua/encoders/slack.lua (hunk 1 above)
Comment (nathanleiby) [inline lua/encoders/slack.lua]: made if configurable. (d748e3e)

File: lua/encoders/slack.lua (hunk 2 above)
Comment (nathanleiby) [inline lua/encoders/slack.lua]: rebase d748e3e

File: lua/encoders/slack.lua [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    config_fields = {"icon_emoji", "username", "channel"}
+    for _, field in ipairs(config_fields) do
+        val = read_config(field)
+        if val then
+          slack_alert[field] = val
+        end
+    end
+
+    -- Read Slack message text from message Fields "msg"
+    text = read_message("Fields[msg]")
```
Comment (mohit) [inline lua/encoders/slack.lua]: is it possible to configure the field where we pull this out of? Maybe something like text_field = "" and default to `Payload`? Curious what you think?

File: lua/encoders/slack.lua [hunk 4]
```diff
@@ -0,0 +1,48 @@
+require "os"
+require "string"
+require "table"
+
+local cjson = require("cjson")
+
+-- Encodes a message to be passed to Slack, using HTTP output
```
Comment (mohit) [inline lua/encoders/slack.lua]: :trophy: documentation!

File: lua/encoders/slack.lua (hunk 3 above)
Comment (nathanleiby) [inline lua/encoders/slack.lua]: Yes, I like that

Comment (nathanleiby) [inline lua/encoders/slack.lua]: b087014

**Comment by nathanleiby** [comment]
offline lgtm from @mohit

**Pull request merged**